packages/datadog-shimmer/src/shimmer.js

```diff
@@ -22,13 +20,12 @@ function copyProperties (original, wrapped) {
 }
 
 function wrapFunction (original, wrapper) {
-  if (typeof original === 'function') assertNotClass(original)
+  assertFunction(original)
```

Comment:
This fails in our tests. Seems like sometimes we pass in undefined (mariadb in case there is no callback. That could be optimized so that we just skip the check in our instrumentation in that case) and other spots. Sometimes we also just pass through `null` directly. All of these should not use shimmer for these cases in the first place. So it's good to detect these cases now, we just have to fix the code accordingly. Question is: separate the behavior / validation from this PR? I guess that is best.

Reply:
Fixed in latest rev